pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/GenericRecord.java

```diff
@@ -61,4 +64,29 @@ default Object getField(Field field) {
      */
     Object getField(String fieldName);
 
+    /**
+     * Return the schema tyoe.
+     *
+     * @return the schema type
+     * @throws UnsupportedOperationException if this feature is not implemented
+     * @see SchemaType#AVRO
+     * @see SchemaType#PROTOBUF_NATIVE
+     * @see SchemaType#JSON
+     */
+    default SchemaType getSchemaType() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Return the internal native representation of the Record,
+     * like a AVRO GenericRecord.
+     * You have to pass the type you would like to obtain.
+     * This method will return null if such type is not supported.
+     *
+     * @return the internal representation of the record, or null if the requested information is not available.
+     */
+    default <T> T getNativeRecord(Class<T> clazz) {
```

Comment:
```suggestion
    default <T> T getNativeRecord() {
```

Why do we need `Class<T>` here?

Reply:
A couple of reasons:
- I am adding this specification because this way the user won't have to deal with ClassCastExceptions or to use 'instanceof'.
- Also the implementation of GenericRecord will be allowed to implement some "compatibility" feature.
For instance if in the future we will move away from JsonNode but the client code still expects to receive a JsonNode, we will be able to return a properly crafted JsonNode instance and to not break clients.
I am not saying we will do this soon, but if we do not add such support we won't be able to add it in the future.

Reply:
I am not sure about that. Having `Class<T> clazz` means that the users need to know what exactly class that each  schema type will use. This will be exactly the same as what I comment on https://github.com/apache/pulsar/pull/9614#issuecomment-784921400. There is no difference between these two approaches.

Because people anyway need to know what exactly class to use, I will instead suggest keeping the interface as simple as `Object getNativeRecord()`. Returning `Object` is very commonly seen in a lot of data processing engines that are dealing with different object types. I don't see it is a big problem to do the same thing here.